### PR TITLE
Remove extra layer of abstraction

### PR DESCRIPTION
**Pull request by jla524**
#698

### CODE REVIEW

**Comment by geohot** [comment]
Err, I don't actually want to rename it.

**Comment by geohot** [comment]
I'm working on something similar, I wonder how much overlap there is? https://github.com/geohot/tinygrad/pull/708

File: tinygrad/runtime/lib.py [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
 import numpy as np
-from typing import TypeVar, Type
+from typing import TypeVar, Type, Optional, Tuple
 from tinygrad.helpers import DType, dtypes, prod, GlobalCounters
 
 _T = TypeVar("_T")
-class RawBuffer:  # pylint: disable=abstract-method
+class RawBuffer:
   def __init__(self, size:int, dtype:DType):
+    self.shape: Optional[Tuple[int, ...]] = None
```
Comment (geohot) [inline tinygrad/runtime/lib.py:9]: This definitely shouldn't have a shape. Shape is tracked in the LazyBuffer

File: disassemblers/adreno/__init__.py [hunk 2]
```diff
@@ -10,7 +10,7 @@ def disasm(buf):
     shared = pathlib.Path(__file__).parent / "disasm.so"
     if not os.path.isfile(shared):
       os.system(f'cd {pathlib.Path(__file__).parent} && gcc -shared disasm-a3xx.c -o disasm.so')
-    fxn = ctypes.CDLL(shared)['disasm']
+    fxn = ctypes.CDLL(str(shared))['disasm']
```
Comment (geohot) [inline disassemblers/adreno/__init__.py:13]: hmm, I think there's a method for this? curious why you need it

**Comment by geohot** [comment]
This wasn't what I was thinking, I really want to remove the extra shapetracker, not just the indirection. My PR is looking like -100 lines

File: disassemblers/adreno/__init__.py (hunk 2 above)
Comment (jla524) [reply disassemblers/adreno/__init__.py:13]: I got an error from mypy saying that the parameter should be a string and not Path

**Comment by jla524** [comment]
> This wasn't what I was thinking, I really want to remove the extra shapetracker, not just the indirection. My PR is looking like -100 lines.
> 
> You picked the hardest of the issues :)
> 
> #693 is probably pretty doable.

Sounds good. I'll close this PR then.

**Pull request closed without merging**